shell/platform/common/cpp/text_input_model.h

```diff
@@ -9,7 +9,55 @@
 #include <memory>
 #include <string>
 
+#include "flutter/fml/logging.h"
+
 namespace flutter {
+
+// A directional range of text.
+//
+// A |TextRange| describes a range of text with |base| and |extent| positions.
+// In the case where |base| == |extent|, the range is said to be collapsed, and
+// when |base| > |extent|, the range is said to be reversed.
+class TextRange {
+ public:
+  TextRange(const TextRange&) = default;
+  explicit TextRange(size_t position) : base_(position), extent_(position) {}
+  TextRange(size_t base, size_t extent) : base_(base), extent_(extent) {}
```

Comment:
This can be `explicit` as well; with C++11 and newer, it's useful for multi-argument constructors as well.

Reply:
Ah good point -- I'm assuming due to initialiser lists effectively allowing for mutli-arg version of the single-arg implicit conversion issue.